chainntnfs/txnotifier.go

```diff
@@ -1548,13 +1583,14 @@ func (n *TxNotifier) filterTx(tx *btcutil.Tx, blockHash *chainhash.Hash,
 		notifyDetails := func(confRequest ConfRequest) {
 			Log.Debugf("Found initial confirmation of %v: "+
 				"height=%d, hash=%v", confRequest,
-				blockHeight, blockHash)
+				blockHeight, block.Hash())
 
 			details := &TxConfirmation{
 				Tx:          tx.MsgTx(),
-				BlockHash:   blockHash,
+				BlockHash:   block.Hash(),
 				BlockHeight: blockHeight,
 				TxIndex:     uint32(tx.Index()),
+				Block:       block.MsgBlock(),
```

Comment:
I think we might also need to set this in `interface.go/ConfDetailsFromTxIndex()`, `bitcoind.go/confDetailsManually()`, `btcd.go/confDetailsManually()` and `neutrino.go/historicalConfDetails()`, unless I'm misunderstanding something.

Reply:
Nice catch! Just fixed those areas, will also tighten up the unit tests to ensure they're caught.